framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -894,7 +892,11 @@ public AnnotationFormatter getAnnotationFormatter() {
      *         empty set if no qualifiers are supported
      */
     public final Set<Class<? extends Annotation>> getSupportedTypeQualifiers() {
-        return supportedQuals;
+        if (this.supportedQuals.isEmpty()) {
+            supportedQuals.addAll(createSupportedTypeQualifiers());
+            checkSupportedQuals();
+        }
+        return Collections.unmodifiableSet(supportedQuals);
```

Comment:
suggest you do `supportedQuals = Collections.unmodifiableSet(supportedQuals)` then `return supportedQuals`. The set of qualifiers shouldn't change after initializing the ATF.